src/main/java/uk/gov/register/serialization/CommandParser.java

```diff
@@ -75,17 +80,18 @@ public void addCommand(String s) {
                 break;
             case "append-entry":
                 if (parts.length == 4) {
-                    Entry entry = new Entry(0, stripPrefix(parts[2]), Instant.parse(parts[1]), parts[3]);
+                    Entry entry = new Entry(0, new HashValue(HashingAlgorithm.SHA256, stripPrefix(parts[2])), Instant.parse(parts[1]), parts[3]);
                     entries.put(position++, entry);
-                    updateItemHashCount(entry.getSha256hex());
+                    updateItemHashCount(entry.getSha256hex().getValue());
```

Comment:
You should not need to use `.getvalue()` here. Can we change `itemHashToEntryCount` to be a `HashMap<HashValue, Integer>`. If you have implemented `HashValue.hashCode()` correctly this should work the same.